File: testdata/jennies/rawtypes/field_with_struct_with_defaults/PHPRawTypes/src/Defaults/DefaultsStructComplexFieldNested.php
<?php

namespace Grafana\Foundation\Defaults;

class DefaultsStructComplexFieldNested implements \JsonSerializable
{
    public string $nestedVal;

    /**
     * @param string|null $nestedVal
     */
    public function __construct(?string $nestedVal = null)
    {
        $this->nestedVal = $nestedVal ?: "";
    }

    public function equals(mixed $other): bool
    {
        if (!($other instanceof self)) {
            return false;
        }
    
        if ($this->nestedVal !== $other->nestedVal) {
            return false;
        }
    
        return true;
    }

    /**
     * @param array<string, mixed> $inputData
     */
    public static function fromArray(array $inputData): self
    {
        /** @var array{nestedVal?: string} $inputData */
        $data = $inputData;
        return new self(
            nestedVal: $data["nestedVal"] ?? null,
        );
    }

    /**
     * @return mixed
     */
    public function jsonSerialize(): mixed
    {
        $data = new \stdClass;
        $data->nestedVal = $this->nestedVal;
        return $data;
    }
}

File: testdata/jennies/rawtypes/field_with_struct_with_defaults/JavaRawTypes/defaults/DefaultsStructComplexFieldNested.java
package defaults;

import java.util.Objects;

public class DefaultsStructComplexFieldNested {
    public String nestedVal;
    public DefaultsStructComplexFieldNested() {
        this.nestedVal = "";
    }
    public DefaultsStructComplexFieldNested(String nestedVal) {
        this.nestedVal = nestedVal;
    }

    @Override
    public boolean equals(Object other) {
        if (this == other) return true;
        if (!(other instanceof DefaultsStructComplexFieldNested)) return false;
        DefaultsStructComplexFieldNested o = (DefaultsStructComplexFieldNested) other;
        if (!Objects.equals(this.nestedVal, o.nestedVal)) return false;
        return true;
    }

    @Override
    public int hashCode() {
        return Objects.hash(this.nestedVal);
    }
}
